idaes/core/solvers/petsc.py

```diff
@@ -0,0 +1,626 @@
+        self.stub = stub
+        self.vis_dir = vis_dir
+        self.path = pth
+        self._read()
+        if delete_on_read:
+            self.delete_files()
+
+    def _read(self):
+        with open(f'{self.stub}.col') as f:
+            names = list(map(str.strip, f.readlines()))
+        with open(f'{self.stub}.typ') as f:
+            typ = list(map(int,f.readlines()))
+        self.vars = [name for i, name in enumerate(names) if typ[i] in [0,1]]
+        (t, v, names) = PetscBinaryIOTrajectory.ReadTrajectory(
+            "Visualization-data"
+        )
+        self.time = t
+        self.vecs_by_time = v
+        self.vecs = dict.fromkeys(self.vars, None)
+        for k in self.vecs.keys():
+            self.vecs[k] = [0]*len(self.time)
+        self.vecs["_time"] = list(self.time)
+        for i, v in enumerate(self.vars):
+            for j, vt in enumerate(self.vecs_by_time):
+                self.vecs[v][j] = vt[i]
+
+    def _var_name_list(self, vars):
+        if isinstance(vars, str):
+            vars = [vars]
+        if hasattr(vars, "ctype"):
+            if issubclass(vars.ctype, pyo.Var):
+                vars = [vars]
+        vars = list(map(str, vars))
+        for name in vars:
+            if name not in self.vars:
+                raise RuntimeError(f"Variable {name} not found.")
+        return vars
+
+    def get_vec(self, var):
+        """Return the time vector for var.
+
+        Args:
+            var (str or Var): Variable to get vector for.
+
+        Retruns:
+            (list): time vector
+
+        """
+        var = str(var)
+        return self.vecs[var]
+
+    def get_dt(self):
+        """Get a list of time steps
+
+        Args:
+            None
+
+        Returns:
+            (list)
+        """
+        dt = [None]*(len(self.time) - 1)
+        for i in range(len(self.time)-1):
+            dt[i] = self.time[i + 1] - self.time[i]
+        return dt
+
+    def interpolate_vecs(self, times):
+        """Create a new vector dictionary interpolated at times.
+
+        Args:
+            vars (list): list of variables or variable names to plot.  The
+                varibles include the time index of the final time.
+
+        Returns:
+            (dict)
+        """
+        vecs = dict.fromkeys(self.vars, None)
+        vecs["_time"] = copy.copy(times)
+        for k in vecs.keys():
+            vecs[k] = [0]*len(times)
+        j = 0
+        for i, t in enumerate(times):
+            while t > self.time[j] and j < len(self.time) - 1:
+                j += 1
+            if j == 0:
+                j = 1
+            dt = self.time[j] - self.time[j - 1]
+            w1 = 1 - (t - self.time[j - 1])/dt
+            w2 = (t - self.time[j - 1])/dt
+            for k, v in self.vecs.items():
+                vecs[k][i] = w1*v[j - 1] + w2*v[j]
+        return vecs
+
+    def plot_trajectory_original(self, vars):
+        """Plot function privided by PETSc.
```

Comment:
privided -> provided

Reply:
Due to the scaling issues, I decided to remove this function.  We can do our own plots pretty easy once the trajectory is read in.